src/libltfs/ltfs_internal.c

```diff
@@ -982,6 +982,48 @@ int _ltfs_make_lost_found(tape_block_t ip_eod, tape_block_t dp_eod,
 	return 0;
 }
 
+/**
+ * Find the appropriate append address for a partition at which calling ltfs_write_index() will
+ * correctly restore consistency to the partition.
+ *
+ * @param dev The ltfs device to work on
+ * @param part The partition to examine
+ * @param block The starting block of the final index on the partition
+ * @return 0 if eod is the appropriate append address, >0 for the absolute block address to append
+ *         at, <0 on error
+ */
+static int _ltfs_find_append_blk_after_idx(struct device_data *dev, tape_partition_t part, tape_block_t block) {
+	unsigned int n_fm_after = 0;
+	struct tc_position idx_pos;
+	struct tc_position final_fm_pos;
+	int ret = 0;
+
+	idx_pos.partition = part;
+	idx_pos.block = block;
+	check_err(tape_seek(dev, &idx_pos), 11020E, out);
```

Comment:
Our message checker detects a argument mismatch in this line. Actually `check_err()` macro is defined as below. 

``` C
#define check_err(cmd, msgid, label) \
	do { \
		ret = (cmd); \
		if (ret < 0) { \
			ltfsmsg(LTFS_ERR, msgid, ret); \
			goto label; \
		} \
	} while (0)
```

So the line is expanded to 

```C
	do {
		ret = tape_seek(dev, &idx_pos); 
		if (ret < 0) { 
			ltfsmsg(LTFS_ERR, 11020E, ret);
			goto label;
		}
	} while (0)
```

But `11020E` does't take any arguments like below.

```
		11020E:string { "Cannot mount volume: seek to index failed on the data partition." }
```

You need to add new massage for this purpose or add `%d` into the message and modify all other lines that are using `11020E`.

Reply:
I see, that macro is inappropriate for that message id. I'll fix that.

Reply:
Additionally, 11020E is only the appropriate message for the data partition. I'm fixing this up so we emit 11023E for the index partition as appropriate.